### PR TITLE
Update to 2.3.7

### PR DESCRIPTION
**Pull request by lapineige**
## Problem
- This app is not up to date

## Solution
- Upgrade to 2.3.7

## PR Status
- [x] Code finished.
- [ ] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : JimboJoe
- [x] **Approval (LGTM)** : JimboJoe
- [x] **Approval (LGTM)** : Maniack C
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/wallabag2_ynh%20PR61%20(Official_fork)/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by JimboJoe** [comment]
Did you try it?
I tried it a couple of weeks ago, and LDAP wasn't working any more without any more clue on why :cry:

**Comment by lapineige** [comment]
No I didn't have the time.

I'll take a look.

**Comment by lapineige** [comment]
Just tested an upgrade: it works.

**Comment by lapineige** [comment]
Now I tried directly with version 2.3.7, it works too.

@JimboJoe could you try again just in case ?

**Comment by JimboJoe** [comment]
SSO seems to work for me, but part of the custom patches fails to get applied...

```
77055 DEBUG + for p in $YNH_CWD/../sources/patches/${source_id}-*.patch
77057 DEBUG + patch -p1
77059 DEBUG (Stripping trailing CRs from patch; use --binary to disable.)
77060 DEBUG patching file app/config/services.yml
77064 DEBUG (Stripping trailing CRs from patch; use --binary to disable.)
77065 DEBUG patching file app/config/security.yml
77066 DEBUG Hunk #1 FAILED at 13.
77067 DEBUG 1 out of 2 hunks FAILED -- saving rejects to file app/config/security.yml.rej
77068 DEBUG + for p in $YNH_CWD/../sources/patches/${source_id}-*.patch
77069 DEBUG + patch -p1
77070 DEBUG (Stripping trailing CRs from patch; use --binary to disable.)
77072 DEBUG patching file app/config/security.yml
77074 DEBUG Hunk #1 succeeded at 54 (offset -6 lines).
77075 DEBUG (Stripping trailing CRs from patch; use --binary to disable.)
77080 DEBUG patching file app/config/services.yml
77081 DEBUG (Stripping trailing CRs from patch; use --binary to disable.)
77082 DEBUG patching file src/Wallabag/YunoHostBundle/Security/LogoutSuccessHandler.php
77083 DEBUG + for p in $YNH_CWD/../sources/patches/${source_id}-*.patch
77084 DEBUG + patch -p1
77085 DEBUG patching file vendor/friendsofsymfony/oauth-server-bundle/Storage/OAuthStorage.php
77086 DEBUG Hunk #1 succeeded at 167 (offset 1 line).
```

**Comment by lapineige** [comment]
Hum, I never took a look at those patches, do you know how they are working ?
We don't have any extra information about what's not working ?

**Comment by JimboJoe** [comment]
The patch part that fails relates to LDAP authentication... Did you try using the Firefox plugin and the Android application?

**Comment by lapineige** [comment]
Oh no I did not (this is my test instance, not the main one, so I didn't try that).
I can't test the android app by the way.

So we need to figure out why this patch doesn't work.

**Comment by lapineige** [comment]
Ok, the Firefox Extension works, as well as android app: https://forum.yunohost.org/t/official-app-wallabag2/3966/10

**Comment by JimboJoe** [comment]
I've commited patches update: no more patching problems.
Upgraded my personal instance: working OK, including Firefox plugin and Android app :+1:

**Comment by lapineige** [comment]
Let's merge then ?

**Comment by JimboJoe** [comment]
Nope, let's respect the workflow for official apps :wink:

File: manifest.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     },
     "url": "https://www.wallabag.org",
     "license": "MIT",
-    "version": "2.3.2-1",
+    "version": "2.3.7",
```
Comment (JimboJoe) [inline manifest.json]:
```suggestion
    "version": "2.3.7~ynh1",
```

Comment (lapineige) [reply manifest.json]: What does that change ?

**Comment by lapineige** [comment]
(does this category still exist ? :yum:)

This is just merging into testing, not the master. And for a minor version, waiting for almost 3 months now.
We'll do that for the testing->master merge.

File: manifest.json (hunk 1 above)
Comment (JimboJoe) [reply manifest.json]: You respect [official packaging rules](https://github.com/YunoHost/doc/blob/master/packaging_apps_manifest.md) :stuck_out_tongue:

**Comment by JimboJoe** [comment]
If you have suggestions regarding the workflow, you can propose them to the community. In the meanwhile, they're are applicable rules... Let's keep it the democratic way :wink:

**Comment by lapineige** [comment]
So we need to call for an additional round of testing from users ?

I installed that upgrade on 2 different servers (in 2 different ways), you did it too, CI is succeeding, some users reported to me that was working… And yet it is not a merge to master :smile:
(I kind of want to get rid of those patches that are lying around and asking maintainers to think about coming back later and taking care of them all the time, ask you can see ^^)

**Comment by JimboJoe** [comment]
I understand your frustration for often living it. Yet, I changed the code this morning, and it hasn't been tested by anyone except me. Furthermore, merging into master will always require reviews.
And you really don't want to experience too often the breaking of multiple users instance due to your merging :fearful:

**Comment by lapineige** [comment]
> and it hasn't been tested by anyone except me

I just did it :)

> And you really don't want to experience too often the breaking of multiple users instance due to your merging :fearful: 

Yeah, clearly, but that counts for master, people using testing know that this might break there instance, and that their own risks.
And in that case, that allow them to use 2.3.7, and maybe test fail2ban with it.

> Furthermore, merging into master will always require reviews.

Yeah, obviously. I just mean I don't feel like testing require that much precaution. (by the way I didn't know we had the same validation process for testing as for master)

Let's wait for more testers then :)

**Comment by JimboJoe** [comment]
That will be better than that. I propose you wait for the approval and merging into testing of this PR, #65 and #67 (requires apps group validation), so it then will have much sense to test all these PR together in a testing release!

**Comment by maniackcrudelis** [comment]
Globally agree with JimboJoe.

Can be merged in 3 days.

File: manifest.json (hunk 1 above)
Comment (JimboJoe) [reply manifest.json]: Don't forget to commit that change :wink:

**Comment by nicofrand** [comment]
Sorry, I did not have time to test it before, so I just upgraded by wallabag version on my YNH with the testing branch, I had no issue, everything worked as expected! Thx

**Comment by lapineige** [comment]
So that counts for #65 too, great :)

**Pull request merged**